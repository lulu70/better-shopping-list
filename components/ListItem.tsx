import React from 'react';
import { StyleSheet, Text, View } from 'react-native';

import { type ItemWithId } from '../App';
import theme from '../constants/theme';
import { horizontalScale, verticalScale } from '../helpers/scaleHelpers';

interface Props {
  item: ItemWithId;
}

const ListItem = ({ item }: Props) => {
  return (
    <View style={styles.listItem} key={item.id}>
      <Text style={styles.checkedIcon}>{item.checked ? '✔️' : '◯'}</Text>
      <Text>{item.content}</Text>
    </View>
  );
};

export default ListItem;

const styles = StyleSheet.create({
  listItem: {
    flexDirection: 'row',
    marginTop: verticalScale(theme.spacing.spacing_6),
  },
  checkedIcon: { marginRight: horizontalScale(theme.spacing.spacing_12) },
});
