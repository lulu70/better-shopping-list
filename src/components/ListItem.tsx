import React from 'react';
import { StyleSheet, TextInput, View } from 'react-native';

import AppButton from './AppButton';
import CheckIcon from '../Icons/CheckIcon';
import RadioButtonIcon from '../Icons/RadioButtonIcon';
import TrashIcon from '../Icons/TrashIcon';
import theme from '../constants/theme';
import MainContext from '../context/MainContext/MainContext';
import SearchContext from '../context/SearchContext/SearchContext';
import { horizontalScale, verticalScale } from '../helpers/scaleHelpers';
import { type ItemWithId } from '../screens/Main';

interface Props {
  item: ItemWithId;
}

const ListItem = ({ item }: Props) => {
  const { changeCheckedItem, deleteItem, editItem } =
    React.useContext(MainContext);
  const { isSearching } = React.useContext(SearchContext);
  const [itemContent, setItemContent] = React.useState(item.content);

  const handleItemChange = (text: string) => {
    setItemContent(text);
  };
  const handleInputBlur = () => {
    if (itemContent !== item.content) {
      editItem(item.id, itemContent);
    }
  };
  return (
    <View style={styles.listItem} key={item.id}>
      <AppButton
        onPress={() => {
          changeCheckedItem(item.id);
        }}
        disabled={isSearching}
      >
        {item.checked ? <CheckIcon /> : <RadioButtonIcon />}
      </AppButton>
      <TextInput
        value={itemContent}
        onChangeText={handleItemChange}
        style={[
          styles.textInput,
          {
            color: isSearching
              ? theme.colors.text_disabled
              : theme.colors.text_black,
          },
        ]}
        editable={!isSearching}
        onBlur={handleInputBlur}
        multiline
      />
      <AppButton
        onPress={() => {
          deleteItem(item.id);
        }}
        disabled={isSearching}
      >
        <TrashIcon />
      </AppButton>
    </View>
  );
};

export default ListItem;

const styles = StyleSheet.create({
  listItem: {
    flexDirection: 'row',
    marginTop: verticalScale(theme.spacing.spacing_12),
    paddingVertical: verticalScale(theme.spacing.spacing_8),
    alignItems: 'center',
  },
  textInput: {
    flex: 1,
    fontSize: horizontalScale(theme.fontSize.fontSize_18),
    lineHeight: verticalScale(theme.fontSize.fontSize_20),
    marginHorizontal: horizontalScale(theme.spacing.spacing_12),
  },
});
