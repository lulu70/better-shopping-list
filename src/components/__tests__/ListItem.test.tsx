import { fireEvent, render, screen } from '@testing-library/react-native';

import ListItem from '../ListItem';

const mockChangeCheckedItem = jest.fn();
const mockDeleteItem = jest.fn();
const mockEditItem = jest.fn();
const mockGoIntoEditMode = jest.fn();
const mockGetOutOfEditMode = jest.fn();
const mockMainContextReturnValue = {
  changeCheckedItem: mockChangeCheckedItem,
  deleteItem: mockDeleteItem,
  editItem: mockEditItem,
  goIntoEditMode: mockGoIntoEditMode,
  getOutOfEditMode: mockGetOutOfEditMode,
  itemInEditMode: null,
  inEditMode: false,
};
const mockSearchContextReturnValue = {
  isSearching: false,
};

const mockScrollToTop = jest.fn();
const mockScrollToItem = jest.fn();

const sampleItem = {
  id: '1631067373',
  checked: false,
  content: 'bread',
  updatedAt: 1631067373,
};

jest.mock('react', () => ({
  ...jest.requireActual('react'),
  useContext: jest
    .fn()
    .mockImplementationOnce(() => mockMainContextReturnValue)
    .mockImplementationOnce(() => mockSearchContextReturnValue)
    .mockImplementationOnce(() => mockMainContextReturnValue)
    .mockImplementationOnce(() => mockSearchContextReturnValue)
    .mockImplementationOnce(() => mockMainContextReturnValue)
    .mockImplementationOnce(() => ({
      isSearching: true,
    }))
    .mockImplementationOnce(() => mockMainContextReturnValue)
    .mockImplementationOnce(() => mockSearchContextReturnValue)
    .mockImplementationOnce(() => mockMainContextReturnValue)
    .mockImplementationOnce(() => mockSearchContextReturnValue)
    .mockImplementationOnce(() => mockMainContextReturnValue)
    .mockImplementationOnce(() => mockSearchContextReturnValue)
    .mockImplementationOnce(() => ({
      ...mockMainContextReturnValue,
      itemInEditMode: sampleItem,
      inEditMode: true,
    }))
    .mockImplementationOnce(() => mockSearchContextReturnValue)
    .mockImplementationOnce(() => ({
      ...mockMainContextReturnValue,
      itemInEditMode: sampleItem,
      inEditMode: true,
    }))
    .mockImplementationOnce(() => mockSearchContextReturnValue)
    .mockImplementationOnce(() => ({
      ...mockMainContextReturnValue,
      itemInEditMode: sampleItem,
      inEditMode: true,
    }))
    .mockImplementationOnce(() => mockSearchContextReturnValue)
    .mockImplementationOnce(() => ({
      ...mockMainContextReturnValue,
      itemInEditMode: sampleItem,
      inEditMode: true,
    }))
    .mockImplementationOnce(() => mockSearchContextReturnValue),
}));

beforeEach(() => {
  jest.clearAllMocks();
});
describe('ListItem', () => {
  it('should match snapshot unchecked', () => {
    render(
      <ListItem
        item={sampleItem}
        scrollToTop={mockScrollToTop}
        scrollToItem={mockScrollToItem}
      />,
    );
    expect(screen.toJSON()).toMatchSnapshot();
  });
  it('should match snapshot checked', () => {
    render(
      <ListItem
        item={{ ...sampleItem, checked: true }}
        scrollToTop={mockScrollToTop}
        scrollToItem={mockScrollToItem}
      />,
    );
    expect(screen.toJSON()).toMatchSnapshot();
  });
  it('should match snapshot in searching state', () => {
    render(
      <ListItem
        item={sampleItem}
        scrollToTop={mockScrollToTop}
        scrollToItem={mockScrollToItem}
      />,
    );
    expect(screen.toJSON()).toMatchSnapshot();
  });
  it('should check the item', () => {
    render(
      <ListItem
        item={sampleItem}
        scrollToTop={mockScrollToTop}
        scrollToItem={mockScrollToItem}
      />,
    );
    const radioButtonIcon = screen.getByTestId('RadioButtonIcon');
    fireEvent.press(radioButtonIcon);
    expect(mockChangeCheckedItem).toHaveBeenCalledWith(sampleItem.id);
  });
  it('should delete the item', () => {
    render(
      <ListItem
        item={sampleItem}
        scrollToTop={mockScrollToTop}
        scrollToItem={mockScrollToItem}
      />,
    );
    const radioButtonIcon = screen.getByTestId('TrashIcon');
    fireEvent.press(radioButtonIcon);
    expect(mockDeleteItem).toHaveBeenCalledWith(sampleItem.id);
  });
  it('should go into edit mode', () => {
    render(
      <ListItem
        item={sampleItem}
        scrollToTop={mockScrollToTop}
        scrollToItem={mockScrollToItem}
      />,
    );
    const contentWrapper = screen.getByText(sampleItem.content);
    fireEvent.press(contentWrapper);
    expect(mockScrollToItem).toHaveBeenCalledWith(sampleItem);
    expect(mockGoIntoEditMode).toHaveBeenCalledWith(sampleItem);
    expect(screen.toJSON()).toMatchSnapshot();
  });
  it('should edit the item', async () => {
    const newContent = 'bread and butter';
    render(
      <ListItem
        item={sampleItem}
        scrollToTop={mockScrollToTop}
        scrollToItem={mockScrollToItem}
      />,
    );
    const textInput = screen.getByDisplayValue(sampleItem.content);
    fireEvent.changeText(textInput, newContent);
    fireEvent(textInput, 'blur');
    expect(mockScrollToTop).toHaveBeenCalled();
    expect(mockEditItem).toHaveBeenCalledWith(sampleItem.id, newContent);
  });
  it('should not edit if theres no change', async () => {
    render(
      <ListItem
        item={sampleItem}
        scrollToTop={mockScrollToTop}
        scrollToItem={mockScrollToItem}
      />,
    );
    const textInput = screen.getByDisplayValue(sampleItem.content);
    fireEvent.changeText(textInput, sampleItem.content);
    fireEvent(textInput, 'blur');
    expect(mockEditItem).not.toHaveBeenCalled();
  });
});
