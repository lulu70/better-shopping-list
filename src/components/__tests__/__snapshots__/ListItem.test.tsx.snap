// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem should go into edit mode 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": false,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    hitSlop={10}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "opacity": 1,
        },
        {
          "marginRight": 24,
        },
      ]
    }
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight="52"
      bbWidth="52"
      focusable={false}
      height={52}
      meetOrSlice={0}
      minX={0}
      minY={-960}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "flex": 0,
            "height": 52,
            "width": 52,
          },
        ]
      }
      testID="RadioButtonIcon"
      vbHeight={960}
      vbWidth={960}
      width={52}
    >
      <RNSVGGroup
        fill={
          {
            "payload": 4278190080,
            "type": 0,
          }
        }
      >
        <RNSVGPath
          d="M480.134-120q-74.673 0-140.41-28.339-65.737-28.34-114.365-76.922-48.627-48.582-76.993-114.257Q120-405.194 120-479.866q0-74.673 28.339-140.41 28.34-65.737 76.922-114.365 48.582-48.627 114.257-76.993Q405.194-840 479.866-840q74.673 0 140.41 28.339 65.737 28.34 114.365 76.922 48.627 48.582 76.993 114.257Q840-554.806 840-480.134q0 74.673-28.339 140.41-28.34 65.737-76.922 114.365-48.582 48.627-114.257 76.993Q554.806-120 480.134-120ZM480-160q134 0 227-93t93-227q0-134-93-227t-227-93q-134 0-227 93t-93 227q0 134 93 227t227 93Zm0-320Z"
          fill={
            {
              "payload": 4278190080,
              "type": 0,
            }
          }
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": false,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    hitSlop={0}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "opacity": 1,
        },
        {
          "flex": 1,
          "paddingVertical": 26.285714285714285,
        },
      ]
    }
  >
    <Text
      style={
        {
          "fontSize": 36,
          "lineHeight": 32.857142857142854,
        }
      }
    >
      bread
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": false,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    hitSlop={10}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "opacity": 1,
        },
        {},
      ]
    }
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight="52"
      bbWidth="52"
      fill="#dc3545"
      focusable={false}
      height={52}
      meetOrSlice={0}
      minX={0}
      minY={-960}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "flex": 0,
            "height": 52,
            "width": 52,
          },
        ]
      }
      testID="TrashIcon"
      vbHeight={960}
      vbWidth={960}
      width={52}
    >
      <RNSVGGroup
        fill={
          {
            "payload": 4292621637,
            "type": 0,
          }
        }
        propList={
          [
            "fill",
          ]
        }
      >
        <RNSVGPath
          d="M304.615-160q-26.846 0-45.731-18.884Q240-197.769 240-224.615V-720h-40v-40h160v-30.77h240V-760h160v40h-40v495.385Q720-197 701.5-178.5 683-160 655.385-160h-350.77ZM680-720H280v495.385q0 10.769 6.923 17.692T304.615-200h350.77q9.23 0 16.923-7.692Q680-215.385 680-224.615V-720ZM392.307-280h40.001v-360h-40.001v360Zm135.385 0h40.001v-360h-40.001v360ZM280-720v520-520Z"
          fill={
            {
              "payload": 4278190080,
              "type": 0,
            }
          }
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
</View>
`;

exports[`ListItem should match snapshot checked 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": false,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    hitSlop={10}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "opacity": 1,
        },
        {
          "marginRight": 24,
        },
      ]
    }
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight="52"
      bbWidth="52"
      focusable={false}
      height={52}
      meetOrSlice={0}
      minX={0}
      minY={-960}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "flex": 0,
            "height": 52,
            "width": 52,
          },
        ]
      }
      testID="CheckIcon"
      vbHeight={960}
      vbWidth={960}
      width={52}
    >
      <RNSVGGroup
        fill={
          {
            "payload": 4278190080,
            "type": 0,
          }
        }
      >
        <RNSVGPath
          d="M382-240 154-468l57-57 171 171 367-367 57 57-424 424Z"
          fill={
            {
              "payload": 4278190080,
              "type": 0,
            }
          }
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": false,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    hitSlop={0}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "opacity": 1,
        },
        {
          "flex": 1,
          "paddingVertical": 26.285714285714285,
        },
      ]
    }
  >
    <Text
      style={
        {
          "fontSize": 36,
          "lineHeight": 32.857142857142854,
        }
      }
    >
      bread
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": false,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    hitSlop={10}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "opacity": 1,
        },
        {},
      ]
    }
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight="52"
      bbWidth="52"
      fill="#dc3545"
      focusable={false}
      height={52}
      meetOrSlice={0}
      minX={0}
      minY={-960}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "flex": 0,
            "height": 52,
            "width": 52,
          },
        ]
      }
      testID="TrashIcon"
      vbHeight={960}
      vbWidth={960}
      width={52}
    >
      <RNSVGGroup
        fill={
          {
            "payload": 4292621637,
            "type": 0,
          }
        }
        propList={
          [
            "fill",
          ]
        }
      >
        <RNSVGPath
          d="M304.615-160q-26.846 0-45.731-18.884Q240-197.769 240-224.615V-720h-40v-40h160v-30.77h240V-760h160v40h-40v495.385Q720-197 701.5-178.5 683-160 655.385-160h-350.77ZM680-720H280v495.385q0 10.769 6.923 17.692T304.615-200h350.77q9.23 0 16.923-7.692Q680-215.385 680-224.615V-720ZM392.307-280h40.001v-360h-40.001v360Zm135.385 0h40.001v-360h-40.001v360ZM280-720v520-520Z"
          fill={
            {
              "payload": 4278190080,
              "type": 0,
            }
          }
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
</View>
`;

exports[`ListItem should match snapshot in searching state 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": true,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    hitSlop={10}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "opacity": 0.2,
        },
        {
          "marginRight": 24,
        },
      ]
    }
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight="52"
      bbWidth="52"
      focusable={false}
      height={52}
      meetOrSlice={0}
      minX={0}
      minY={-960}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "flex": 0,
            "height": 52,
            "width": 52,
          },
        ]
      }
      testID="RadioButtonIcon"
      vbHeight={960}
      vbWidth={960}
      width={52}
    >
      <RNSVGGroup
        fill={
          {
            "payload": 4278190080,
            "type": 0,
          }
        }
      >
        <RNSVGPath
          d="M480.134-120q-74.673 0-140.41-28.339-65.737-28.34-114.365-76.922-48.627-48.582-76.993-114.257Q120-405.194 120-479.866q0-74.673 28.339-140.41 28.34-65.737 76.922-114.365 48.582-48.627 114.257-76.993Q405.194-840 479.866-840q74.673 0 140.41 28.339 65.737 28.34 114.365 76.922 48.627 48.582 76.993 114.257Q840-554.806 840-480.134q0 74.673-28.339 140.41-28.34 65.737-76.922 114.365-48.582 48.627-114.257 76.993Q554.806-120 480.134-120ZM480-160q134 0 227-93t93-227q0-134-93-227t-227-93q-134 0-227 93t-93 227q0 134 93 227t227 93Zm0-320Z"
          fill={
            {
              "payload": 4278190080,
              "type": 0,
            }
          }
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": true,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    hitSlop={0}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "opacity": 0.2,
        },
        {
          "flex": 1,
          "paddingVertical": 26.285714285714285,
        },
      ]
    }
  >
    <Text
      style={
        {
          "fontSize": 36,
          "lineHeight": 32.857142857142854,
        }
      }
    >
      bread
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": true,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    hitSlop={10}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "opacity": 0.2,
        },
        {},
      ]
    }
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight="52"
      bbWidth="52"
      fill="#dc3545"
      focusable={false}
      height={52}
      meetOrSlice={0}
      minX={0}
      minY={-960}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "flex": 0,
            "height": 52,
            "width": 52,
          },
        ]
      }
      testID="TrashIcon"
      vbHeight={960}
      vbWidth={960}
      width={52}
    >
      <RNSVGGroup
        fill={
          {
            "payload": 4292621637,
            "type": 0,
          }
        }
        propList={
          [
            "fill",
          ]
        }
      >
        <RNSVGPath
          d="M304.615-160q-26.846 0-45.731-18.884Q240-197.769 240-224.615V-720h-40v-40h160v-30.77h240V-760h160v40h-40v495.385Q720-197 701.5-178.5 683-160 655.385-160h-350.77ZM680-720H280v495.385q0 10.769 6.923 17.692T304.615-200h350.77q9.23 0 16.923-7.692Q680-215.385 680-224.615V-720ZM392.307-280h40.001v-360h-40.001v360Zm135.385 0h40.001v-360h-40.001v360ZM280-720v520-520Z"
          fill={
            {
              "payload": 4278190080,
              "type": 0,
            }
          }
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
</View>
`;

exports[`ListItem should match snapshot unchecked 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": false,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    hitSlop={10}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "opacity": 1,
        },
        {
          "marginRight": 24,
        },
      ]
    }
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight="52"
      bbWidth="52"
      focusable={false}
      height={52}
      meetOrSlice={0}
      minX={0}
      minY={-960}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "flex": 0,
            "height": 52,
            "width": 52,
          },
        ]
      }
      testID="RadioButtonIcon"
      vbHeight={960}
      vbWidth={960}
      width={52}
    >
      <RNSVGGroup
        fill={
          {
            "payload": 4278190080,
            "type": 0,
          }
        }
      >
        <RNSVGPath
          d="M480.134-120q-74.673 0-140.41-28.339-65.737-28.34-114.365-76.922-48.627-48.582-76.993-114.257Q120-405.194 120-479.866q0-74.673 28.339-140.41 28.34-65.737 76.922-114.365 48.582-48.627 114.257-76.993Q405.194-840 479.866-840q74.673 0 140.41 28.339 65.737 28.34 114.365 76.922 48.627 48.582 76.993 114.257Q840-554.806 840-480.134q0 74.673-28.339 140.41-28.34 65.737-76.922 114.365-48.582 48.627-114.257 76.993Q554.806-120 480.134-120ZM480-160q134 0 227-93t93-227q0-134-93-227t-227-93q-134 0-227 93t-93 227q0 134 93 227t227 93Zm0-320Z"
          fill={
            {
              "payload": 4278190080,
              "type": 0,
            }
          }
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": false,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    hitSlop={0}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "opacity": 1,
        },
        {
          "flex": 1,
          "paddingVertical": 26.285714285714285,
        },
      ]
    }
  >
    <Text
      style={
        {
          "fontSize": 36,
          "lineHeight": 32.857142857142854,
        }
      }
    >
      bread
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": false,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    hitSlop={10}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "opacity": 1,
        },
        {},
      ]
    }
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight="52"
      bbWidth="52"
      fill="#dc3545"
      focusable={false}
      height={52}
      meetOrSlice={0}
      minX={0}
      minY={-960}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "flex": 0,
            "height": 52,
            "width": 52,
          },
        ]
      }
      testID="TrashIcon"
      vbHeight={960}
      vbWidth={960}
      width={52}
    >
      <RNSVGGroup
        fill={
          {
            "payload": 4292621637,
            "type": 0,
          }
        }
        propList={
          [
            "fill",
          ]
        }
      >
        <RNSVGPath
          d="M304.615-160q-26.846 0-45.731-18.884Q240-197.769 240-224.615V-720h-40v-40h160v-30.77h240V-760h160v40h-40v495.385Q720-197 701.5-178.5 683-160 655.385-160h-350.77ZM680-720H280v495.385q0 10.769 6.923 17.692T304.615-200h350.77q9.23 0 16.923-7.692Q680-215.385 680-224.615V-720ZM392.307-280h40.001v-360h-40.001v360Zm135.385 0h40.001v-360h-40.001v360ZM280-720v520-520Z"
          fill={
            {
              "payload": 4278190080,
              "type": 0,
            }
          }
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
</View>
`;
